tests/integration/offer/test_benefit.py

```diff
@@ -1,25 +1,46 @@
-from django.test import TestCase
+import pytest
+
 from django.utils import six
 
 
 from oscar.apps.offer.models import Benefit
 from oscar.test import factories
 
 
-class TestBenefitProxyModels(TestCase):
+@pytest.fixture
+def range():
+    return factories.RangeFactory()
+
+
+@pytest.mark.django_db
+class TestBenefitProxyModels(object):
+    """
+
+    https://docs.djangoproject.com/en/1.11/topics/db/models/#proxy-models
 
-    def test_name_and_description(self):
+    """
+
+    def test_name_and_description(self, range):
         """
         Tests that the benefit proxy classes all return a name and
         description. Unfortunately, the current implementations means
         a valid range is required.
         This test became necessary because the complex name/description logic
         broke with the python_2_unicode_compatible decorator.
         """
-        range = factories.RangeFactory()
         for type, __ in Benefit.TYPE_CHOICES:
             benefit = Benefit(type=type, range=range)
-            self.assertTrue(all([
+            assert all([
                 benefit.name,
                 benefit.description,
-                six.text_type(benefit)]))
+                six.text_type(benefit)])
+
+    def test_proxy(self, range):
+        for type, __ in Benefit.TYPE_CHOICES:
```

Comment:
The only thing - can we not use reserved names for Python builtins? E.g. `type` -> `type_choice`.

Thanks.